.github/workflows/build+test+deploy.yml

```diff
@@ -55,45 +55,39 @@ jobs:
     - name: Setup .NET
       uses: actions/setup-dotnet@v4
       with:
-        dotnet-version: ${{ env.DOTNET_VERSION }}
+        global-json-file: global.json
     - name: Restore tools
       run: dotnet tool restore
     - name: Build
-      run: dotnet fsi build.fsx
+      shell: pwsh
+      run: ./build.ps1
     - name: Pack
-      run: dotnet fsi build.fsx -t Pack
+      shell: pwsh
+      run: ./build.ps1 DotnetPack
     - name: Get Changelog Entry
       id: changelog_reader
       uses: mindsers/changelog-reader-action@v1
       with:
         version: ${{ github.ref }}
         path: ./CHANGELOG.md
-    - name: Upload binaries to nuget (if nugetKey present)
+    - name: Publish (if tag)
+      if: startsWith(github.ref, 'refs/tags/')
+      shell: pwsh
       env:
-        nuget-key: ${{ secrets.NUGET_KEY }}
+        NUGET_TOKEN: ${{ secrets.NUGET_TOKEN }}
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: dotnet fsi build.fsx -t Push
-    - name: Create Release (if tag)
-      if: startsWith(github.ref, 'refs/tags/')
-      id: create_release
-      uses: actions/create-release@latest
+        FAKE_DETAILED_ERRORS: true
+
+      run: ./build.ps1 Publish
+
+    - name: PublishToGitHub (if master branch)
+      if: github.ref == 'refs/heads/master'
```

Comment:
@xperiandri this logic was moved in the past from GithubActions to F#, let's keep it in F# please